webpack/dev.config.js

```diff
@@ -6,73 +6,96 @@ import webpack from 'webpack';
+      hot: true,
+      stats: {
+        assets: true,
+        colors: true,
+        version: false,
+        hash: false,
+        timings: true,
+        chunks: false,
+        chunkModules: false
       }
-    ],
-    loaders: [
-      {
-        test: /\.json$/,
-        loader: 'json'
-      },
-      {
-        test: /\.(jpe?g|png|gif|svg|woff|eot|ttf)$/,
-        loader: 'url?limit=10000&name=[sha512:hash:base64:7].[ext]'
-      },
-      {
-        test: /\.js$|.jsx$/,
-        exclude: /node_modules/,
-        loaders: ['react-hot', 'babel?experimental']
-      },
-      {
-        test: /\.scss$/,
-        loader: 'style!css?sourceMap!autoprefixer?browsers=last 2 version!sass?outputStyle=expanded&sourceMap'
-      }
-    ]
+    }
   },
-  plugins: [
-
-    // hot reload
-    new webpack.HotModuleReplacementPlugin(),
-    new webpack.NoErrorsPlugin(),
-
-    new webpack.DefinePlugin({
-      'process.env': {
-        BROWSER: JSON.stringify(true),
-        NODE_ENV: JSON.stringify('development')
-      }
-    }),
-
-    new webpack.optimize.DedupePlugin(),
-    new webpack.optimize.OccurenceOrderPlugin(),
-
-    function () { this.plugin('done', writeStats); },
-    function () { this.plugin('done', startKoa); }
-
-  ],
-  resolve: {
-    extensions: ['', '.js', '.json', '.jsx'],
-    modulesDirectories: ['node_modules', 'app']
+  webpack: {
+    devtool: 'eval-source-map',
+    entry: {
+      app: [
+        `webpack-dev-server/client?http://localhost:${WEBPACK_PORT}`,
+        'webpack/hot/only-dev-server',
+        './app/index.js'
+      ]
+    },
+    publicPath: PUBLIC_PATH,
+    output: {
+      path: path.join(__dirname, '../dist'),
+      filename: '[name]-[chunkhash].js',
+      publicPath: PUBLIC_PATH
+    },
+    module: {
+      preLoaders: [
+        {
+          test: /\.js$|.jsx$/,
+          exclude: /node_modules/,
+          loaders: ['eslint', 'jscs']
+        }
+      ],
+      loaders: [
+        {
+          test: /\.json$/,
+          loader: 'json'
+        },
+        {
+          test: /\.(jpe?g|png|gif|svg|woff|eot|ttf)$/,
+          loader: 'url?limit=10000&name=[sha512:hash:base64:7].[ext]'
+        },
+        {
+          test: /\.js$|.jsx$/,
+          exclude: /node_modules/,
+          loaders: ['react-hot', 'babel?experimental']
+        },
+        {
+          test: /\.scss$/,
+          loader: 'style!css?sourceMap!autoprefixer?browsers=last 2 version!sass?outputStyle=expanded&sourceMap'
+        }
+      ]
+    },
+    plugins: [
+  
+      // hot reload
+      new webpack.HotModuleReplacementPlugin(),
+      new webpack.NoErrorsPlugin(),
+  
+      new webpack.DefinePlugin({
+        'process.env': {
+          BROWSER: JSON.stringify(true),
+          NODE_ENV: JSON.stringify('development')
+        }
+      }),
+  
+      new webpack.optimize.DedupePlugin(),
+      new webpack.optimize.OccurenceOrderPlugin(),
+  
+      function () { this.plugin('done', writeStats); },
```

Comment:
We can remove this line, since it's gonna use `webpackConfig.server.port` with `koa-proxy` now?